### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by tagliala**
```
codespell **/*.md --skip="node_modules/**/*" -w
codespell **/*.rb --skip="node_modules/**/*" -L afile,filetest,noes -w
```

## Summary by CodeRabbit

- **Documentation**
  - Corrected typos in troubleshooting, upgrade, and React documentation for better clarity and correctness.
  
- **Tests**
  - Updated a test case description for accuracy and improved readability.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -180,7 +180,7 @@ default: &default
...
 
-Note, if you are exposing globals, like jQuery, to non-webpack dependencies (like an inline script) via the [expose-loader](https://webpack.js.org/loaders/expose-loader/), you will need to override the default of `defer: true` to be `defer:false` your call to the `javascript_pack_tag` so that the browser will load your bundle to setup the global variable before other code depends on it. However, you really should try to remove the dependendency on such globals.
+Note, if you are exposing globals, like jQuery, to non-webpack dependencies (like an inline script) via the [expose-loader](https://webpack.js.org/loaders/expose-loader/), you will need to override the default of `defer: true` to be `defer:false` your call to the `javascript_pack_tag` so that the browser will load your bundle to setup the global variable before other code depends on it. However, you really should try to remove the dependency on such globals.
```
[the remark this answers is not in the document]
Comment (tagliala) [reply docs/troubleshooting.md:183]: Should I apply this correction?

[the remark this answers is not in the document]
Comment (tagliala) [reply docs/troubleshooting.md:183]: Waiting for a human confirmation, `setup` also have other occurrences like

- `# This script is a way to setup or update your development environment automatically.`
- `The below information applies to a React on Rails app, except that you need to setup some environment variables as follows:`

and I guess it is quite commonly misused

[the remark this answers is not in the document]
Comment (justin808) [reply docs/troubleshooting.md:183]: yes, fix all

**Review by justin808 — CHANGES_REQUESTED** [review]

**Comment by tagliala** [comment]
Fixed `to setup` => `to set up`. Sorry I'm not a native language speaker, so in other contexts it looks like it is a noun (or can be a noun, like "Setup with yalc" or "easy setup")

File: docs/react.md [hunk 2]
```diff
@@ -3,9 +3,10 @@
 These steps describe creating a Rails/React app, using Shakapacker as the bundler.
 
 ## Easy Setup
+
 If you'd like easy integration of React with Ruby on Rails, see [React on Rails](https://github.com/shakacode/react_on_rails).
 
-The below information applies to a React on Rails app, except that you need to setup some environment variables as follows:
+The following information applies to a React on Rails app. Additionally, you need to set up some environment variables as follows:
 
 ```
```
[the remark this answers is not in the document]
Comment (tagliala) [reply docs/react.md:11]: @coderabbitai is this infinite recursion?

**Review by tomdracz — APPROVED** [review]

**Pull request merged**